src/txn/txn_ckpt.c

```diff
@@ -2273,6 +2273,9 @@ __wt_checkpoint_tree_reconcile_update(WT_SESSION_IMPL *session, WT_TIME_AGGREGAT
             ckpt->run_write_gen = btree->run_write_gen;
             WT_TIME_AGGREGATE_COPY(&ckpt->ta, ta);
         }
+
+    /* Ensure the maximum timestamp used for reconciliation is tracked. */
+    btree->rec_max_timestamp = WT_MAX(ta->newest_start_durable_ts, ta->newest_stop_durable_ts);
```

Comment:
Assigning it directly can overwrite a higher value that is already existing. I agree to overwrite it during RTS or recovery. But in normal scenarios, only when it is lower, it can be updated.

Scenario:
1. Checkpoint has finished part of the btree.
2. Later before it finishes the root page info here, eviction writes a page to the disk and updates the `rec_max_timestamp`.
3. Replacing it with the checkpoint information can overwrite a higher value.

Reply:
Thank you for the feedback - I did not think about this scenario.